dfm_sentence_trf/main.py

```diff
@@ -43,37 +43,25 @@ def finetune(
 ):
     raw_config = Config().from_disk(config_path)
     cfg = registry.resolve(raw_config)
-    sent_trf_kwargs = {}
+    sent_trf_kwargs = dict()
     sent_trf_kwargs["device"] = cfg["model"].get("device", "cpu")
     if cache_folder is not None:
         sent_trf_kwargs["cache_folder"] = cache_folder
-    pooling_kwargs = {}
-    pooling_kwargs["pooling_mode"] = cfg["model"].get("pooling_mode", None)
-    pooling_kwargs["pooling_mode_cls_token"] = cfg["model"].get(
-        "pooling_mode_cls_token", False
-    )
-    pooling_kwargs["pooling_mode_max_tokens"] = cfg["model"].get(
-        "pooling_mode_max_tokens", False
-    )
-    pooling_kwargs["pooling_mode_mean_tokens"] = cfg["model"].get(
-        "pooling_mode_mean_tokens", True
-    )
-    pooling_kwargs["pooling_mode_mean_sqrt_len_tokens"] = cfg["model"].get(
-        "pooling_mode_mean_sqrt_len_tokens", False
-    )
-    training_kwargs = {}
-    training_kwargs["epochs"] = cfg["training"].get("epochs", 5)
-    training_kwargs["warmup_steps"] = cfg["training"].get("warmup_steps", 100)
 
     embedding = models.Transformer(cfg["model"]["base_model"])
     pooling = models.Pooling(
         word_embedding_dimension=embedding.get_word_embedding_dimension(),
-        **pooling_kwargs,
     )
     model = SentenceTransformer(
         modules=[embedding, pooling], **sent_trf_kwargs
     )
-    train_model(model, cfg["tasks"], **training_kwargs)
+
+    epochs = cfg["training"].get("epochs", 5)
+    warmup_steps = cfg["training"].get("warmup_steps", 100)
+    batch_size = cfg["training"].get("batch_size", 30)
```

Comment:
I wouldn't do the default like this. Do it with a default config.

Reply:
Can you provide an example on how you would do it? I'm not very comfortable with defaults in confection just yet.

Reply:
I have tried writing a pydantic schema according to the guide in Thinc's documentation:
```python
from pydantic import BaseModel, StrictInt, StrictStr


class TrainingSchema(BaseModel):
    epochs: StrictInt = 5
    warmup_steps: StrictInt = 100
    batch_size: StrictInt = 30


class ModelSchema(BaseModel):
    name: StrictStr
    base_model: StrictStr
    device: StrictStr = "cpu"


class ConfigSchema(BaseModel):
    model: ModelSchema
    training: TrainingSchema

    class Config:
        extra = "allow"

```

When I try to fill the config though, I get an error:
```python
AttributeError: 'FieldInfo' object has no attribute 'type_'
```.
Maybe `config.merge()` would work?

Reply:
Addressed, default config is now defined in config.py and merged with the loaded config.

Reply:
Exactly the way I would have done it! If the default gets a lot bigger I would write it out to a file that you load in but otherwise exactly the same.